api/src/main/java/ai/djl/nn/convolutional/Conv1dTranspose.java

```diff
@@ -0,0 +1,248 @@
+ * with the License. A copy of the License is located at
+ *
+ * http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+ * OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+package ai.djl.nn.convolutional;
+
+import ai.djl.ndarray.NDArray;
+import ai.djl.ndarray.NDList;
+import ai.djl.ndarray.types.LayoutType;
+import ai.djl.ndarray.types.Shape;
+import ai.djl.nn.Block;
+import ai.djl.util.Preconditions;
+
+/**
+ * A {@code Conv1dTranspose} layer works similar to {@link Deconvolution} layer with the exception
+ * of the number of dimension it operates on being only one, which is {@link LayoutType#WIDTH}. The
+ * channel of the input data may be more than one, depending on what data is processed. Each filter
+ * slides through the data with only one direction of movement along the dimension itself.
+ *
+ * <p>The input to a {@code Conv1dTranspose} is an {@link ai.djl.ndarray.NDList} with a single 3-D
+ * {@link ai.djl.ndarray.NDArray}. The layout of the {@link ai.djl.ndarray.NDArray} must be "NCW".
+ * The shapes are
+ *
+ * <ul>
+ *   <li>{@code data: (batch_size, channel, width)}
+ *   <li>{@code weight: (num_filter, channel, kernel[0])}
+ *   <li>{@code bias: (num_filter,)}
+ *   <li>{@code out: (batch_size, num_filter, out_width)} <br>
+ *       {@code out_width = f(width, kernel[0], pad[0], o_pad[0], stride[0], dilate[0])} <br>
+ *       {@code where f(x, k, p, o_p, s, d) = (x-1)*s-2*p+k+o_p}
+ * </ul>
+ *
+ * <p>Both {@code weight} and {@code bias} are learn-able parameters.
+ */
+public class Conv1dTranspose extends Deconvolution {
+
+    private static final LayoutType[] EXPECTED_LAYOUT = {
+        LayoutType.BATCH, LayoutType.CHANNEL, LayoutType.WIDTH
+    };
+
+    private static final String STRING_LAYOUT = "NCW";
+    private static final int NUM_DIMENSIONS = 3;
+
+    Conv1dTranspose(Builder builder) {
+        super(builder);
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    protected LayoutType[] getExpectedLayout() {
+        return EXPECTED_LAYOUT;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    protected String getStringLayout() {
+        return STRING_LAYOUT;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    protected int numDimensions() {
+        return NUM_DIMENSIONS;
+    }
+
+    /**
+     * Applies 1D deconvolution over an input signal composed of several input planes.
+     *
+     * @param input the input {@code NDArray} of shape (batchSize, inputChannel, width)
+     * @param weight filters {@code NDArray} of shape (outChannel, inputChannel/groups, width)
+     * @return the output of the conv1dtranspose operation
+     */
+    public static NDList conv1dtranspose(NDArray input, NDArray weight) {
```

Comment:
```suggestion
    public static NDList conv1dTranspose(NDArray input, NDArray weight) {
```
same applies to the rest